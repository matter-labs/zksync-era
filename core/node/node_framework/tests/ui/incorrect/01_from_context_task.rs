#![allow(dead_code)]

use zksync_node_framework::{FromContext, Resource, StopReceiver, Task, TaskId};

#[derive(Clone)]
struct ResourceA;

impl Resource for ResourceA {
    fn name() -> String {
        "a".to_string()
    }
}

struct TaskA;

#[async_trait::async_trait]
impl Task for TaskA {
    fn id(&self) -> TaskId {
        "batch_status_updater".into()
    }

    async fn run(self: Box<Self>, _stop_receiver: StopReceiver) -> anyhow::Result<()> {
        Ok(())
    }
}

#[derive(FromContext)]
struct SimpleStruct {
    _field: ResourceA,
    #[context(task)]
    _field_2: TaskA,
}

fn main() {}
