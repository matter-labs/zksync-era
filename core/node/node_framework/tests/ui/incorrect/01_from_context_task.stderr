error: `task` attribute is not allowed in `FromContext` macro
  --> tests/ui/incorrect/01_from_context_task.rs:31:5
   |
31 |     _field_2: TaskA,
   |     ^^^^^^^^
